Frameworks/AddressBook/_ABAddressBook.mm

```diff
@@ -0,0 +1,52 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import "_ABAddressBook.h"
+#import "UWP/WindowsApplicationModelContacts.h"
+
+@interface _ABAddressBook ()
+
+@end
+
+@implementation _ABAddressBook
+
+- (id)init {
+    self = [super init];
+    if (self) {
+        __block BOOL completed = NO;
+        __block WACContactStore* result = nil;
+
+        [WACContactManager requestStoreAsyncWithAccessType:WACContactStoreAccessTypeAllContactsReadOnly
+            success:^(WACContactStore* success) {
+                completed = YES;
+                result = success;
+            }
+            failure:^(NSError* failure) {
+                completed = YES;
+            }];
+
+        // Wait until async method completes.
```

Comment:
To tack on to what Ben said, busy loops are also not my favorite mechanism for waiting. pumping the run loop here is also questionable as you don't know what thread you are being called from. What if you are on a background thread already (that doesn't have a run loop) what does it mean to be pumping the current run loop in that case?

---
In reply to: [73199542](https://github.com/Microsoft/WinObjC/pull/786#discussion_r73199542) [](ancestors = 73199542)